### PR TITLE
Feature/remove sample function

### PR DESCRIPTION
**Pull request by GeorgeNash**
- `sample` taken off `Vertex`
- `Probabilistic` extends `Samplable`
- Can no longer call `sample` on operations

### CODE REVIEW

File: keanu-project/src/main/java/io/improbable/keanu/algorithms/particlefiltering/ParticleFilter.java [hunk 1]
```diff
@@ -132,7 +134,7 @@ private void addObservedVertexToParticles(List<Particle> particles,
     }
 
     private <T> void sampleValueAndAddToParticle(Vertex<T> vertex, Particle particle) {
-        T sample = vertex.sample(random);
+        T sample = ((Probabilistic<T>) vertex).sample(random);
```
Comment (GeorgeNash) [inline keanu-project/src/main/java/io/improbable/keanu/algorithms/particlefiltering/ParticleFilter.java]: This looks unsafe but the `vertex` is coming from a set inside `sort` which extracts only latent variables

File: keanu-project/src/main/java/io/improbable/keanu/network/BayesianNetwork.java [hunk 2]
```diff
@@ -209,13 +210,16 @@ public static void setFromSampleAndCascade(List<? extends Vertex> vertices) {
 
     public static void setFromSampleAndCascade(List<? extends Vertex> vertices, KeanuRandom random) {
```
Comment (gordoncaleb) [inline keanu-project/src/main/java/io/improbable/keanu/network/BayesianNetwork.java]: Can we just take a List<? extends Probabilistic>?

File: keanu-project/src/main/java/io/improbable/keanu/vertices/LogProbGraph.java [hunk 3]
```diff
@@ -47,10 +46,6 @@ public DoubleTensor calculate() {
             return getPlaceholderVertexValue(this);
         }
 
-        @Override
-        public DoubleTensor sample(KeanuRandom random) {
```
Comment (gordoncaleb) [inline keanu-project/src/main/java/io/improbable/keanu/vertices/LogProbGraph.java]: Nice! RIP `sample()`

File: keanu-project/src/main/java/io/improbable/keanu/network/BayesianNetwork.java (hunk 2 above)
Comment (gordoncaleb) [reply keanu-project/src/main/java/io/improbable/keanu/network/BayesianNetwork.java]: Talked about this offline. OK...

**Pull request merged**